underscore.js

```diff
@@ -1288,13 +1288,42 @@
     return type === 'function' || type === 'object' && !!obj;
   };
 
-  // Add some isType methods: isArguments, isFunction, isString, isNumber, isDate, isRegExp, isError, isMap, isWeakMap, isSet, isWeakSet.
-  _.each(['Arguments', 'Function', 'String', 'Number', 'Date', 'RegExp', 'Error', 'Symbol', 'Map', 'WeakMap', 'Set', 'WeakSet'], function(name) {
+
+  /**
+  Helper function for returning the object tag (e.g. [object Object]); implemented
+  to work around IE11 bug (Object.Prototype.toString with Map, Set and WeakMap
+  doesn't return '[object Map]' etc.)
+  **/
+  var returnTag = function(obj) {
+    // If toString returns [object Object] and its constructor is a function,
+    // proceed with comparing the constrcutor's Function.prototype.toString with
+    // Map, WeakMap and WeakSet tags.
+    var objectTag = Object.prototype.toString.call(obj),
+        objCtor = objectTag === '[object Object]' ? obj.constructor : null,
+        objCtorString = typeof objCtor === 'function' ? Function.prototype.toString.call(objCtor) : '';
+
+    if (objCtorString) {
+      var mapCtorString = typeof Map === 'function' ? Function.prototype.toString.call(Map) : '',
+          weakMapCtorString = typeof WeakMap === 'function' ? Function.prototype.toString.call(WeakMap) : '',
+          setCtorString = typeof Set === 'function' ? Function.prototype.toString.call(Set) : '';
+
+      switch (objCtorString) {
+        case mapCtorString : return '[object Map]';
+        case weakMapCtorString : return '[object WeakMap]';
+        case setCtorString : return '[object Set]';
+      }
+    }
+    return objectTag;
+  };
```

Comment:
This is a very large workaround just to fix three `isType` functions for one version of one browser. I don't like the size of it and I don't like the fact that it is being applied to all the other `isType` functions as well.